src/EditorFeatures/CSharpTest/ReplaceDefaultLiteral/ReplaceDefaultLiteralTests.cs

```diff
@@ -0,0 +1,921 @@
+            await TestWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        switch ((0, true)) { case [||]default when true: }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch ((0, true)) { case default((int, bool)) when true: }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Theory]
+        [InlineData("value")]
+        [InlineData("null")]
+        [InlineData("default")]
+        [InlineData("() => { }")]
+        [InlineData("")]
+        public async Task TestCSharp7_1_InCasePatternSwitchLabel_NotForInvalidType(string expression)
+        {
+            await TestMissingWithLanguageVersionsAsync(
+$@"class C
+{{
+    void M()
+    {{
+        switch ({expression}) {{ case [||]default when true: }}
+    }}
+}}", s_csharp7_1above);
+        }
+
+        [Fact]
+        public async Task TestCSharp7_1_InIsPattern_Bool()
+        {
+            await TestWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        if (true is [||]default) { }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (true is false) { }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Fact]
+        public async Task TestCSharp7_1_InIsPattern_Bool_InParentheses()
+        {
+            await TestWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        if (true is ([||]default)) { }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (true is (false)) { }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Fact]
+        public async Task TestCSharp7_1_InIsPattern_Bool_NotInsideCast()
+        {
+            await TestMissingWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        if (true is (bool)[||]default) { }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Fact]
+        public async Task TestCSharp7_1_InIsPattern_Bool_NotOnDefaultExpression()
+        {
+            await TestMissingWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        if (true is [||]default(bool)) { }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Fact]
+        public async Task TestCSharp7_1_InIsPattern_Bool_NotOnFalseLiteral()
+        {
+            await TestMissingWithLanguageVersionsAsync(
+@"class C
+{
+    void M()
+    {
+        if (true is [||]false) { }
+    }
+}", s_csharp7_1above);
+        }
+
+        [Theory]
+        [InlineData("int", "0")]
+        [InlineData("uint", "0U")]
+        [InlineData("byte", "0")]
+        [InlineData("sbyte", "0")]
+        [InlineData("short", "0")]
+        [InlineData("ushort", "0")]
+        [InlineData("long", "0L")]
+        [InlineData("ulong", "0UL")]
+        [InlineData("float", "0F")]
+        [InlineData("double", "0D")]
+        [InlineData("decimal", "0M")]
+        public async Task TestCSharp7_1_InIsPattern_NumericType(string type, string expectedLiteral)
+        {
+            await TestWithLanguageVersionsAsync(
+$@"class C
+{{
+    void M()
+    {{
+        {type} value = 1;
```

Comment:
💡 If you use the following definition for `value`, you can combine this test with some of the ones that follow.

```
{type} value = default({type});
```

It's also possible to combine the tests by making `value` a field or parameter instead of a local.

📝 If you prefer the clarity of separating numeric types from other types, feel free to leave this in its current form.

Reply:
I noticed that but didn't really bother to combine them. I saw combining numeric types into a theory as the main improvement because there was just so many of them. Anyway, you're right that we can make this better, and I should do that.

➡️ Fixed in 4e101c7bbb5721a5ca82e1d0d8969f33d96fc0b3